configure.py

```diff
@@ -2164,8 +2164,9 @@ def make_bin_override():
 
 if options.compile_commands_json:
   gyp_args += ['-f', 'compile_commands_json']
-  os.path.islink('./compile_commands.json') and os.unlink('./compile_commands.json')
-  os.symlink('./out/' + config['BUILDTYPE'] + '/compile_commands.json', './compile_commands.json')
+  if sys.platform != 'win32':
+    os.path.islink('./compile_commands.json') and os.unlink('./compile_commands.json')
```

Comment:
```suggestion
    os.path.isfile('./compile_commands.json') and os.unlink('./compile_commands.json')
```

Reply:
But we are creating `./compile_commands.json` as a symlink in this case (non-Windows), and the original code used to use `islink`. `isfile` actually follows symlinks so can be confusing in this case...

Reply:
I think the original code can be improved.

```
the os.path.isfile(path) method that returns True if the path is a file or a symlink to a file.
```

If you check for `islink()`, if someone creates a file called `compile_commands.json`, it will not remove and overwrite it.

Reply:
Won't `isfile` return `False` if `compile_commands.json` is a symlink that points to a non-existent file?

Reply:
Yeah, that's true. Thanks for the correction, so a combination of `isfile` or `islink` might be better?

Reply:
Maybe `os.path.lexists` so it's one call, if we are afraid of someone creating a file with that name.